pkg/patch/internal/internalfakes/fake_formatter.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by counterfeiter. DO NOT EDIT.
+package internalfakes
+
+import (
+	"sync"
+
+	"k8s.io/release/pkg/patch"
+)
+
+type FakeFormatter struct {
+	MarkdownToHTMLStub        func(string, string) (string, error)
+	markdownToHTMLMutex       sync.RWMutex
+	markdownToHTMLArgsForCall []struct {
+		arg1 string
+		arg2 string
+	}
+	markdownToHTMLReturns struct {
+		result1 string
+		result2 error
+	}
+	markdownToHTMLReturnsOnCall map[int]struct {
+		result1 string
+		result2 error
+	}
+	invocations      map[string][][]interface{}
+	invocationsMutex sync.RWMutex
```

Comment:
are the two mutexes here to conform unit tests?
can they be combined into a single RW mutex if this is for unit tests only?

i may have missed related concurrency in the business logic.

Reply:
This is generated code. I guess most of your concerns about the fakes stem from the fact that counterfeiter is written in a generic way, is able to create fakes for functions & interfaces and does not make assumptions if the thing it fakes is used in concurrent code or not.

Things in `<something>fake` or which are named `fake_<something>.go` are most probably counterfeiter generated fakes, are tagged as being autogenerated (e.g. [here](https://github.com/kubernetes/release/pull/1040/files/48ccfcef8650bf0cac62f561573a55cb3b08c46d#diff-47628a87ff589fa18ea485ba5d8dca18R17)) and thus can be ignored IMHO.

Having said that, if you have concerns with the counterfeiter fakes, let me know and I can see to address them upstream.

Reply:
looks like i missed the `// Code generated by counterfeiter. DO NOT EDIT.` part